src/input/input_json.rs

```diff
@@ -5,26 +5,28 @@ use pyo3::types::PyDict;
 use speedate::MicrosecondsPrecisionOverflowBehavior;
 use strum::EnumMessage;
 
+use crate::data_value::DataValue;
 use crate::errors::{ErrorType, InputValue, LocItem, ValError, ValResult};
 
 use super::datetime::{
     bytes_as_date, bytes_as_datetime, bytes_as_time, bytes_as_timedelta, float_as_datetime, float_as_duration,
     float_as_time, int_as_datetime, int_as_duration, int_as_time, EitherDate, EitherDateTime, EitherTime,
 };
-use super::parse_json::JsonArray;
+
 use super::shared::{float_as_int, int_as_bool, map_json_err, str_as_bool, str_as_int};
 use super::{
     EitherBytes, EitherFloat, EitherInt, EitherString, EitherTimedelta, GenericArguments, GenericIterable,
-    GenericIterator, GenericMapping, Input, JsonArgs, JsonInput,
+    GenericIterator, GenericMapping, Input, JsonArgs,
 };
 
-impl<'a> Input<'a> for JsonInput {
+impl<'a> Input<'a> for DataValue {
     /// This is required by since JSON object keys are always strings, I don't think it can be called
     #[cfg_attr(has_no_coverage, no_coverage)]
     fn as_loc_item(&self) -> LocItem {
         match self {
-            JsonInput::Int(i) => (*i).into(),
-            JsonInput::String(s) => s.as_str().into(),
+            DataValue::Int(i) => (*i).into(),
+            DataValue::String(s) => s.as_str().into(),
+            DataValue::Py(obj) => obj.as_ref(unsafe { Python::assume_gil_acquired() }).as_loc_item(),
```

Comment:
looks like lots of these methods will need  `py` as a paramter to remove the `unsafe`?

Reply:
Yeah